pandas/tests/frame/test_alter_axes.py

```diff
@@ -270,6 +270,31 @@ def test_set_index_raise_on_type(self, frame_of_index_cols, box,
             df.set_index(['A', df['A'], box(df['A'])],
                          drop=drop, append=append)
 
+    def test_set_index_custom_label_type(self):
+        # GH 24969
+
+        class Thing(object):
+            def __init__(self, name, color):
+                self.name = name
+                self.color = color
+
+            def __str__(self):
```

Comment:
Is `__str__` used still?

Reply:
Would work without, just means the repr is ugly (and maybe harder to debug if someone doesn't know what's happening):
```
>>> df = pd.DataFrame({thing1: [0, 1], thing2: [2, 3]})
>>> df
   <__main__.Thing object at 0x000001973BAB9908>  <__main__.Thing object at 0x000001973BAB99B0>
0                                              0                                              2
1                                              1                                              3
```

Reply:
👍. Just (overly) concerned about behavior differences between objects where `__str__` is / isn't defined. Best to include it.